sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/CosmosTemplate.java

```diff
@@ -210,16 +213,30 @@ public <T> T insert(String containerName, T objectToSave) {
      * @return the inserted item
      */
     public <T> T insert(String containerName, T objectToSave, PartitionKey partitionKey) {
+
         Assert.hasText(containerName, "containerName should not be null, empty or only whitespaces");
         Assert.notNull(objectToSave, "objectToSave should not be null");
 
         @SuppressWarnings("unchecked") final Class<T> domainType = (Class<T>) objectToSave.getClass();
-        containerName = getContainerName(domainType);
 
         markAuditedIfConfigured(objectToSave);
         generateIdIfNullAndAutoGenerationEnabled(objectToSave, domainType);
 
-        final JsonNode originalItem = mappingCosmosConverter.writeJsonNode(objectToSave);
+        JsonNode originalItem = mappingCosmosConverter.writeJsonNode(objectToSave);
```

Comment:
I am curious about why are we stripping the fields from the item here in each API instead of moving this logic inside `writeJsonNode` and `readInternal` APIs of MappingCosmosConverter?
We do similar work there in order to handle fields with diff annotations than their names, like for id and etag

Reply:
It defaults to true, so I did not have to change every method that uses it. But its only a small change, and perhaps it should default to false.